src/stores/AccountController.ts

```diff
@@ -176,34 +187,27 @@ export const logout = async () => {
 
     await restoreFavorites();
     await restoreWatchHistory();
+
+    // it's needed for the InPlayer SDK
     await accountService.logout();
   });
 };
 
 export const register = async (email: string, password: string) => {
-  await useConfig(async ({ cleengId, cleengSandbox }) => {
-    const localesResponse = await cleengAccountService.getLocales(cleengSandbox);
-
-    if (localesResponse.errors.length > 0) throw new Error(localesResponse.errors[0]);
-
-    const responseRegister = await cleengAccountService.register(
-      {
-        email: email,
-        password: password,
-        locale: localesResponse.responseData.locale,
-        country: localesResponse.responseData.country,
-        currency: localesResponse.responseData.currency,
-        publisherId: cleengId,
-      },
-      cleengSandbox,
-    );
-
-    if (responseRegister.errors.length) throw new Error(responseRegister.errors[0]);
-
-    await getAccount(responseRegister.responseData);
+  await withAccountService(async ({ accountService, config }) => {
+    useAccountStore.setState({ loading: true });
+    const { auth, user } = await accountService.register({ config, email, password });
+    useAccountStore.setState({
+      auth,
+      user,
+    });
 
-    await updatePersonalShelves();
+    // @todo statement will be removed once the fav and history are done on InPlayer side
+    if (auth.refreshToken) {
+      await updatePersonalShelves();
```

Comment:
Yes, that's what is meant by the todo. The next thing that I am working is the fav and history, so until it's finished this condition will be here because otherwise it throws an error on inplayer side because of cleengId missing.

Reply:
OK, I just thought there might be a cleaner way to stub it so you didn't have to come back to the todo later.  This is temporary, so I won't worry about it.